flake/dev-shell.nix

```diff
@@ -0,0 +1,52 @@
+{
+  perSystem =
+    {
+      pkgs,
+      config,
+      inputs',
+      ...
+    }:
+    {
+      devShells = {
+        default =
+          let
+            check = pkgs.writeShellApplication {
+              name = "stylix-check";
+              runtimeInputs = with pkgs; [
+                nix
+                nix-fast-build
+              ];
+              text = ''
+                cores="$(nproc)"
+                system="$(nix eval --expr builtins.currentSystem --impure --raw)"
+                nix-fast-build \
+                  --eval-max-memory-size 512 \
+                  --eval-workers "$cores" \
+                  --flake ".#checks.$system" \
+                  --no-link \
+                  --skip-cached \
+                  "$@"
+              '';
+            };
+          in
```

Comment:
Formatting nit:

Personally, I'd move this `let in` block up to after the `perSystem` module args; L8-L9.

That way, the `pkgs.mkShell` wouldn't need to be indented:

```nix
default = pkgs.mkShell {
  # ...
};
```